torchgeo/trainers/segmentation.py

```diff
@@ -89,20 +89,29 @@ def __init__(self, **kwargs: Any) -> None:
         self.save_hyperparameters()  # type: ignore[operator]
         self.hyperparams = cast(Dict[str, Any], self.hparams)
 
-        self.ignore_zeros = None if kwargs["ignore_zeros"] else 0
-
+        if not (
+            isinstance(kwargs["ignore_index"], int) or (kwargs["ignore_index"] is None)
+        ):
+            raise ValueError("ignore_index must be an int or None")
+        if (kwargs["ignore_index"] is not None) and (kwargs["loss"] == "jaccard"):
+            warnings.warn(
+                "Warning ignore_index has no effect on training when"
+                + " loss='jaccard'",
```

Comment:
```suggestion
                "ignore_index has no effect on training when loss='jaccard'",
```
The warning will already start with "UserWarning: ..." so no need to warn twice.